test/e2e/versionskew/operator.go

```diff
@@ -0,0 +1,227 @@
+/*
+Copyright 2020 Intel Corporation.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package versionskew
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	api "github.com/intel/pmem-csi/pkg/apis/pmemcsi/v1beta1"
+	"github.com/intel/pmem-csi/test/e2e/deploy"
+	"golang.org/x/net/context"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = deploy.DescribeForSome("versionskew", func(d *deploy.Deployment) bool {
+	// Run these tests only for bare operator deployments, not for the operator
+	// deployed by OLM. Because there is no way how to ask olm to stop the operator
+	// but keep the CRD.
```

Comment:
Isn't that a problem for people using OLM to manage the operator? How are they supposed to do upgrades and downgrades?

Reply:
The operator upgrade is supported by the newer  `operator-sdk` releases for the operators in the 'bundle manifest'(`operator-sdk generate bundle`) format, but not in 'package manifest'. We should anyway move to bundle format as the packagemanifests are becoming obsolete. I am working on this now, will submit a new PR.

But downgrading is not supported by operator-sdk.